module/mempool/herocache/dns_cache.go

```diff
@@ -0,0 +1,143 @@
+package herocache
+
+import (
+	"net"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+)
+
+type DNSCache struct {
+	ipCache  *stdmap.Backend
+	txtCache *stdmap.Backend
+}
+
+func NewDNSCache(sizeLimit uint32, logger zerolog.Logger) *DNSCache {
+	return &DNSCache{
+		txtCache: stdmap.NewBackendWithBackData(
+			herocache.NewCache(sizeLimit, 8, heropool.LRUEjection, logger)),
+		ipCache: stdmap.NewBackendWithBackData(
+			herocache.NewCache(sizeLimit, 8, heropool.LRUEjection, logger)),
+	}
+}
+
+// PutIpDomain adds the given ip domain into the cache.
+func (d *DNSCache) PutIpDomain(domain string, timestamp int64, addresses []net.IPAddr) bool {
+	i := ipEntity{
+		id:        domainToIdentifier(domain),
```

Comment:
This is using 'MakeID'.
That seems like overkill.  What does it gain for you, since the lookups have to use MakeID(), too ??

Reply:
Hashing our offline conversations around this comment:

> At this stage, we needed a collision-resistant identifier for DNS records. We want to make sure that no two distinct domains collide upon storing in the cache. Otherwise, it would cause safety concerns on the code that DNS queries for the domain `A` are resolved to domain `B` only because `A` and `B` collide on the same identifier in the cache. We examined several options but reached the conclusion that we should not compromise on the collision-resistance feature, since it enables DNS poisoning attacks. Hence we confined utilization of hash literally once per record upon inserting it into the cache. The `Rem` method is barely used only when we face an error in resolving a domain, which is an extremely unlikely event in the current configurations. Also, this translation from `string -> flow.Identifier ` is done only once when we insert a domain into the cache.